app/controllers/authors_controller.rb

```diff
@@ -0,0 +1,41 @@
+class AuthorsController < ApplicationController
+  def index
+    @authors= Author.all
+  end
+
+  def new 
+    @author = Author.new
+  end
+
+  def create
+    @author = Author.new(params.require(:author).permit(:first_name,:last_name,:dob,:email))
+    if @author.save
+      redirect_to authors_path
+    else
+      flash[:notice] = @author.errors.full_messages
+      redirect_to new_author_path
+    end
+  
+  end
+
+  def show
+    @author = Author.find(params[:id])
```

Comment:
Create a one method for common code. Like create a method with name find_author and use that.